### PR TITLE
[DEMO] saveourtool projects instead of GitHub projects

### LINKED ISSUE #1677: Stop using GitHub credentials in save-demo

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR closes #1677

### What's done:
* Added `demo` table to `save_demo` db scheme
* Added `Demo` entity, `DemoRepository` and `DemoService`
* Made `save-demo` work with saveourtool projects instead of GitHub projects

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt [hunk 1]
```diff
@@ -53,30 +63,48 @@ class DemoManagerController(
     fun addDemo(
         @PathVariable organizationName: String,
         @PathVariable projectName: String,
-        @RequestBody demoToolRequest: NewDemoToolRequest,
+        @RequestBody demoDto: DemoDto,
         authentication: Authentication,
-    ): Mono<Unit> = demoToolRequest.toMono()
+    ): Mono<EmptyResponse> = blockingToMono {
+        projectService.findByNameAndOrganizationNameAndStatusIn(
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt:69]: what's about `findByNameAndOrganizationNameAndCreatedStatus`

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/domain/ProjectCoordinates.kt [hunk 2]
```diff
@@ -15,4 +15,18 @@ data class ProjectCoordinates(
      * @return ProjectCoordinates as formatted string "{organizationName}/{projectName}"
      */
     override fun toString() = "$organizationName/$projectName"
+
+    /**
+     * @return true if both [organizationName] and [projectName] are empty, false otherwise
+     */
+    fun consideredEmpty() = organizationName.isEmpty() && projectName.isEmpty()
```
Comment (nulls) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/domain/ProjectCoordinates.kt:22]: why not just `isEmpty`?

**Review by nulls — APPROVED** [review]

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/domain/ProjectCoordinates.kt (hunk 2 above)
Comment (sanyavertolet) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/domain/ProjectCoordinates.kt:22]: Problem with Jackson serialization. `isEmpty` is serialized into Boolean variable `Empty`.
Cannot apply `JsonIgnore` annotation as far as `ProjectCoordinates` are in `commonMain`, while Jackson is `jvm`

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt:69]: Oh sure, will fix it

Comment (sanyavertolet) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt]: Done

**Review by Cheshiriks — APPROVED** [review]

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/domain/ProjectCoordinates.kt (hunk 2 above)
Comment (nulls) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/domain/ProjectCoordinates.kt:22]: got it

**Pull request merged**